File: js/bubblesort.js
function bubbleSort(arr) {
  // change current index and the next
  // arrow function practice
  const swap = (array, arrLen, idx) => {
    if (arrLen - 1 > idx) {
      let rem = array[idx + 1];
      array[idx + 1] = array[idx];
      array[idx] = rem;
    }

    return array;
  };

  // for-loop until fix index
  for (let i = arr.length; i > 0; i--) {
    for (let j = 0; j < i; j++) {
      if (arr[j] > arr[j + 1]) {
        swap(arr, i, j);
      }
    }

  }

  return arr;
}

function bubbleSort2(arr) {
  const swap = (array, i, j) => {
    let tmp = array[j];
    array[j] = array[i];
    array[i] = tmp;
  };

  for (let i = arr.length - 1; i > 0; i--) {
    for (let j = 0; j < i; j++) {
      if (arr[j] > arr[j + 1]) {
        swap(arr, j, j + 1);
      }
    }
  }

  return arr;
}

const toSort = [4, 2, 6, 9, 11, 2, 1, 5, 6, 10];
console.log(bubbleSort2(toSort));

File: python/bubble_sort.py
def bubble_sort(input_val):

    for i in range(len(input_val) - 1, 0, -1):
        for j in range(0, i):
            if j == len(input_val) - 1:
                return input_val
            if input_val[j] > input_val[j + 1]:
                swap(input_val, j)

    print(input_val)


def swap(list, idx):
    if len(list) == idx - 1:
        return list

    rem = list[idx]
    list[idx] = list[idx + 1]
    list[idx + 1] = rem


if __name__ == "__main__":
    bubble_sort([3,2,1,5,9,11,20,23,11,2,1,222,111])